packages/xo-web/src/xo-app/dashboard/health/index.js

```diff
@@ -510,8 +510,11 @@ const HANDLED_VDI_TYPES = new Set(['system', 'user', 'ephemeral'])
   const getVifsByMac = createGetObjectsOfType('VIF')
     .pick(
       createCollectionWrapper(
-        createSelector(getVms.filter([({ blockedOperations: ops }) => !('start' in ops || 'start_on' in ops)]), vms =>
-          flatMap(vms, 'VIFs').sort()
+        createSelector(getVms, vms =>
+          flatMap(
+            filter(vms, ({ blockedOperations: ops }) => !('start' in ops || 'start_on' in ops)),
+            'VIFs'
+          ).sort()
```

Comment:
This is still looping twice over the "same" collection. You can loop only once with a `forEach` that both maps and filters all at once.